### ISSUE TITLE
Use of AdamOptimizer during training produces error

### ISSUE THREAD

**Issue by pblouw** [nengo/nengo_dl#11]
Training a model with TensorFlow's Adam optimizer (`tf.train.AdamOptimizer()`), produces a `FailedPreconditionError`. Minimal example:

```
import nengo
import nengo_dl
import tensorflow as tf

with nengo.Network() as net:
    net.config[nengo.Ensemble].neuron_type = nengo.RectifiedLinear()
    inp = nengo.Node(0)
    ens = nengo.Ensemble(100, 1)
    probe = nengo.Probe(ens)
    
values = np.random.uniform(1, 1, size=(100, 1, 1))
inputs = {inp: values}
outputs = {probe: 2 * values}
 
with nengo_dl.Simulator(net) as sim:
    optimizer = tf.train.AdamOptimizer(0.2)
    sim.train(inputs, outputs, optimizer, 1, 'mse')
```

This may have something to do with how the optimizer requires the initialization of new variables for tracking gradient information.
